src/subscription_manager/gui/registergui.py

```diff
@@ -1992,6 +2102,19 @@ def _refresh(self, callback):
         except Exception:
             self.queue.put((callback, None, sys.exc_info()))
 
+    def __unregister_consumer(self, consumer_uuid, server_info):
+        # Method to actually do the unregister bits
+        old_cp = UEPConnection(**server_info)
+        managerlib.unregister(old_cp, consumer_uuid)
+
+
+    def _unregister_consumer(self, consumer_uuid, server_info, callback):
+        try:
+            self.__unregister_consumer(consumer_uuid, server_info)
```

Comment:
May be useful to catch any exceptions inside of __unregister_consumer, and raise some sort of UnregisterException so s-m-gui/firstboot could decide if they want to ignore that specific exception.

Need to verify nothing in the gui is expecting the vague exceptions though. That is a little bit of
a concern, especially if this could be the first network call made by the gui.